### ISSUE TITLE
LOL needs to be in rune page edit mode or else Runebook won't work [BUG - Enhancement] & Other minor things

### ISSUE THREAD

**Issue by PhalanXdarKen** [Soundofdarkness/RuneBook#105]
This is the best LOL software, it only needs a few tweeks for reaching perfection, it is fast enought and have a nice looking UI.

**MAIN PROBLEM:** 

As shown in Images attached, when I create a new runepage, I always test it on training mode. After this I save it normally in Local Runes section and the new page appears in the software list, but When I enter to the champ selection screen I am unable to click on the send button, it shows light off (darker) and it doesn't work.  Only after you click on the (+) BUTTON, runebook detects that you are editing or creating a rune page and after that you can click that send button. Please see photos below. The best and desired situation would be that once in the champion select screen you can simply click on the send button and then you got your rune pages imported asap and ready to use in that one - click without having to do extra clicks in the plus and in the pencil (edit) buttons.

This is not so practical when you want to create a new rune page and you want to use it inmediately after opening the program. As a player you need fast responsiveness from softwares.

Also a rare thing is that when you have an incomplete rune page in the client, there appears a red exclamation symbol and then the send button keeps active sometimes indefinitely. This allows you to create and use runepages instantly???... whatever. [see image 5]

BTW I've been using a software called Rune Changer and it is great tool too, it is not a exe file, it is a "open.bat" file and if I am not using it anymore it's because it is very slow when you want to select a runepage created, you scroll down to see more pages but it lags a lot even if your PC is fast. The good thing about this Rune Changer program is that I never had a problem like the explained above, it takes effect inmediately. That's why Runebook would be the definitive program if you solve this and also maybe taking in consideration the following points:

**SUGGESTED FEATURES AND CORRECTIONS:**

- There's no sense about "favorite" in option menu because you already added an autoselect champion mode. Having this activated will overlap any favorite section in the program everytime you highlight a champ.

- Create an ALL RUNES section for fast browsing when you start the program.

- Program's auto champ-recognition function don't recognizes champions in any screen before champion select (pre-game screen). What if I am looking at my champs in my "collection" ? it neither works in RUNES SECTION  in "collection" it would be useful to create rune pages from those client sections and not only in champ selection screen automatically using the auto recognition system you've already implemented. Of course you can still do this manually... but is not ideal - fast.

- Item pages and summoner's spells importing - exporting from runebook UI. I know that LOL aready have a itempage creator but it would be even better if we can have everything in just one place and send that data inmediately with just one button. I thing there's a program called Mobalytics that already lets you do this but it is very slow too.

- Add a discord invitation button to the UI into configuration wheel, this is a modern and great place to share all info about the program and writing announcements, even fast and better than github. Remember most users are non developers and only players like me, a lot of ppl don't know how to use Github and prefer discord.

That's all I can say, this is my feedback and a bug reporting, thanks for reading.

![BUG](https://user-images.githubusercontent.com/88815060/150661531-5e2c2d70-862d-435b-9f7f-c34dd79b3653.png)

![bug 2 1](https://user-images.githubusercontent.com/88815060/150661987-42e1d8ac-58d5-4284-9fa7-0222089caf21.png)

![bug 2](https://user-images.githubusercontent.com/88815060/150662018-26badc61-c4c1-4d53-98b0-5d8e76ecf4a5.png)

![bug 3](https://user-images.githubusercontent.com/88815060/150662033-f850f419-0e1e-44d3-b02b-a21331c84f41.png)

![bug 4](https://user-images.githubusercontent.com/88815060/150662163-2ac80bcc-2310-45fe-9b3d-81081774eea6.png)

**Comment by 11Tuvork28** [comment]
Thanks for the feedback and suggestions. As for the issues you describe I can't reproduce them, are you on the newest version? and itemsets can already be exported although only the blitzgg.

**Comment by PhalanXdarKen** [comment]
Yeah @11Tuvork28 I have the latest version. And yes I was taking a look to that new function but the item images doesn't appear yet and it's not so aesthetic nor practical yet. I think UI is very important nowadays. Try enter editing rune mode, this problem occurs generally when I first start the program. It's very hard to explain but the easiest solution could be to make the SEND button always enabled, also I guess that in the script of the program there should be a editing mode dependence and that causes this bug, you can link the SEND button action to the API of LOL similar to "Rune changer". Please check it out: https://runechanger.stirante.com/

**Comment by Soundofdarkness** [comment]
Hey !
First of all, thanks for all the suggestions! 
To mention the issue report, from what I can tell this keeps happening because you currently have one of the default rune-pages selected. We can't overwrite them afaik, and since we otherwise just overwrite the current page (to make sure to not accidentally overwrite a still needed page) and for the ones being not yet fully set up, I'm not sure why we cant see those. I will see if I can look into this.

To the other suggestions:
1) Favorite in this case is mostly to quickly import the rune page to your locally saved pages, while still being able to update it, so I think this is still kind of needed.
2) An all-runes page might be useful, but will probably going to require a nearly full rework of the UI, so I'm not sure on that so far, but maybe at some point.
3) Champions not being selected from collections is something I'm not sure if we can add this (need to check if the client actually sends that at all), so I'm not sure if we can do that either.
4) Itempages being visible in a better way mostly ties in to the rework UI part I think, might be something to do at some point, but will require significant effort. Summoner spells might be doable, but for now probably in the same way as items, but I'm not quite sure if this is needed, due to most of it coming down to preference.
5) Discord is something I can look into, but can't make promises yet.

So mostly stuff we can look into, but at least for my part, I can't make any promises, but those are certainly nice ideas !
Thank you for the time to write all of those down !
(I will leave the issue open for further discussions, since its certainly worth looking into)

**Comment by PhalanXdarKen** [comment]
thank you for taking the time of aswering me. Iam sure that making those changes will improve the app to the next level, A good idea should be make some marketing or invitations to popular streamers who can live talk about the app. I would like to help you with UI but iam sorry that Idk nothing about programming. I will be waiting for changes and updates, keep it going!

**Comment by 11Tuvork28** [comment]
I don't know about that but I'm looking into fixing some bugs I found and making favorite champs a thing, well finishing up #42. 
Also I'm working on finding out the remaining endpoints for the other sites, mainly lolalytics but no idea if and when I get all this done because I don't have that much time currently...

**Comment by Soundofdarkness** [comment]
No problem ! About the marketing part, I don't think that's a possibility for us, simply due to sponsoring content in some way or another (or even marketing in general) usually needs a company, which would not be feasible for now. Additionally it would add to the expectation of runebook permanently being up to date and working perfectly, which is something that's really hard to guarantee as an open-source side project.
And don't worry, simply writing us your ideas and opinions is really valuable as well, so you certainly helped !

@11Tuvork28 Nice ! Don't worry about the time, if you are able to finish what you are working on its just as welcome a while later, and even if it doesn't work out in the end its still nice that you attempted it !

**Comment by 11Tuvork28** [comment]
Hi @PhalanXdarKen you mentioned that the UI could be better, I have been working on the UI for bit specifically the item set page.
I would like to here your thoughts about it although keep in mind that this is just prototyping
![Screenshot_109](https://user-images.githubusercontent.com/59368718/157084002-c4fac213-b694-424f-af78-fc7323bc370b.png)

**Comment by PhalanXdarKen** [comment]
@11Tuvork28 Hi! Iam really impressed that you and your amazing team keeps on work and listen to your audience! ☺️ I shared the program with 10 friends and they are so satisfied and thankful. Some of them are having the same issue of the darker unclickable send button, being unable to import into client if they are not editing a rune page. I will share the software soon with more ppl because I will start streaming ☺️. Now please let me explain some points:

The UI design of the items it's cool, combines well with the software visual style and I don't see anything wrong from a aesthetic sight. I think the important thing about this is that when you hover the Mouse pointer over any item a small window appear with the current original description that appears into game, showing cost, combinable items like the mythicals, stats provided, etc. 

Another important point would be that user can create himself extra sections clicking a + plus button same like client item page creating. The problem with the client item creation tool is that it is not so comfortable to use, too much clicks for reaching that section and I have to say it is ugly. So it would be faster and more practical editing your item sets into Runebook and so user can edit them even if game isn't open. Same would apply to summoner spells. That means user can click on another send button to import that into client, I have seen that another softwares import all of these data in the champ selection just before game starts. In short terms, spells and items edited from the software and runes edited from client but saved and with the existing import function from runebook. Usually for item sets I create like 5 different sections, sometimes you want to add situational items, AP builds, etc. But with runebook maybe users will prefer to create item pages for every champion due to the confort it provides same like runes, appearing as a list and with just one click import. 

Lastly the most hard idea would be doing the same that a function of "rune changer" software provides. They have inserted a button with their icon just at the side of the pencil icon in champ selection, that way players can simply click there and a small window appears, there you can scroll down and select which runepage you want to use, but in the case of runebook, this function would open not only the runepage selection, but also the itemsets and summoner spells. Please download as a test and analysis purpose that rune changer  it's very good too and you can learn a lot from that, they have a website too. As I mentioned in the post above, I don't use it because it is not as complete as runebook is and it is very slow, so bad optimization and consumes a lot of Ram. Maybe you can talk about these points with your teammates and at least some of the upgrades will be of their like. I hope this helps. 

In the stramings the ppl will ask me about what software Iam using and of course I will say Runebook, I hope that a lot of donations happens, because you deserve them. 💪🏻☺️ thank you!

**Comment by 11Tuvork28** [comment]
@PhalanXdarKen Thanks for  the feedback!! 
But let me clarify that I'm not affiliated with the 'team' behind this project and that I'm just a random person working on this because I have fun.

So I'm working on a few points you mentoined and I will orient myself at your points. Currently I'm working a few new features and UI revamp also adding a new tab with info about the selected champ:
- Counter pick
- Skill order
- Info about the spells

As for the features I wanna implent(in order that I will work on these):
- Add op.gg to item set exporter
- Auto accept match
- favorite champions
- auto select favorite champ for the assigned lane
- Runepage/Itemset creator

Maybe I will look into making a overlay but yeah probably not any time soon, also I won't be working any longer on bringing this to typescript due to having a bad time with electron...

I think I also figured out the bug you and some of your friends are experiencing, I will be trying to fix it.
And thanks again!!!

**Comment by PhalanXdarKen** [comment]
Hi @11Tuvork28  again 😊 so much thank you for your efforts, I really use runebook everyday its my favorite lol app. I like the idea of the skill order, about that it will be a great idea if the user can create multiple "skillpages", this way you can select which one to use in every match. Now, what other apps does is that they show an enlighted square mark around the skill that you should level up, this can really be tricky because sometimes you level up another skill so the markers should be adaptable according to the current level of all the skills. In resume: runebook could show square markers for level up suggestion into game according to the selected skill page you have created before in runebook ( the same way as if you were creating rune pages and importing them with a just one click button).

I really recommend you to download and test "Mobalytics" app, I don't use it because it is not customizable, the users can only follow stright pro builds and it's not my like because most users and me now tends to create our own builds. But you can analyze the behavior of that skill square markers and how they works into game. I don't know any other apps that does this better than Mobalytics. They also give the option of in game tabs - small windows showing scores and other stuff but again I fell them not so comfortable for the sight. 

In conclusion Runebook has the best potential of being the better and ultimate customizable lol app, where you can create your own builds via different pages from every champ in the game. The most important is already done, the basic program already works and now just remains new functions to be added. This is just an opinion from a user of lots of apps and games, so I give my feedback according of my needs and what I see is working better right now both for me and other players. Iam not programmer but I have notions of graphic design and UI. Keep it going bro! You are the best 😄

**Comment by Soundofdarkness** [comment]
I'm just going to quickly try to respond as well, but it's going to be quite short since I'm ill at the moment and don't really feel well enough to do much, so apologies for that.
First of all, thank you for all the feedback !
I can't really speak from the feature part myself, since I'm not working much on those myself anymore, but I can just say that if people like @11Tuvork28 want to work on those features I'm gladly going to release them, I just hope that we are actually going to be able to maintain those, since all of those seem to be quite complex.
On the point of the overlay I'm personally very torn since those are a huge effort to create them in a proper and performant way (since for electron the main way is having a click through window on top with the potential need of passing through click events in javascript to the game behind it) which takes a ton of resources from my limited testing (and needs hardware acceleration disabled additionally). So I can offer to try around a bit with that since I've done that for private projects in the past , but can't say if we will be able to implement that.
The rest sounds doable if @11Tuvork28 wants to work on those.

And just a small mention about the team part: This project so far has no real persistent team, just awesome volunteers providing new features or fixing old bugs and me just coming in every week or two to see if something needs to be released. So I'd say if anyone counts as a team then its the persons adding/fixing things ! (Which is why im trying to always add everyone who worked on something in the release notes 😄 ) 

I hope the above made some sense and thanks for all the work and suggestions everyone !

**Comment by 11Tuvork28** [comment]
@soundofdarkness First of all I hope you get well soon and no need to apologize for that!
Thanks for merging my PRs, I got a few more PRs on the way this weekend specifically for:
* Images for the items and price as shown in an screenshot above
* op.gg item sets
* Auto accept a match
* More locals and localized Item names

But I don't think I get them all ready today, so they might not be viable for the coming release. On that note if you or anyone wants to see what I'm working on I merge my work in the experimental branch of my fork so check it out if you want.

Also I want try to build all those complex features in a way that is decoupled from the plugins and therefore from changing parts, so maintaining should be doable.

**Comment by PhalanXdarKen** [comment]
Of course ! Please take your time and thank you two for keeping in mind and alive this project. I hope you get better and I think @11Tuvork28 is going to make a great work. I forgot to say that there's is a little bug, Op.gg runes isn't showing in runebook suggested runes,  there appears a sad face, all the other tabs shows well like for example blitz.gg or u.gg. Take the time you need for any uptades, if you need a translator English to LAS Spanish, you can write me here: michaelreyk@gmail.com it's my personal email, maybe I can bring you the latam spanish texts and translate some words or descriptions, etc. I can only help with that and with my ideas / suggestions. I use runebook everyday, so I know it well. Again thanks you two and good luck 🤞

**Comment by Soundofdarkness** [comment]
@11Tuvork28 Thank you and sounds good ! I've replied on the PR for the auto accept (TLDR: Need to ask riot first since I so far dont have permission to do more than runes, and that is sadly required by Riot for Client API projects), but the rest should be fine good! 
@PhalanXdarKen Thank you ! And about the OP.GG bug the fix is technically already merged, just not released yet, but that should happen within a day or two. And thank you for asking with translations, that might be an offer we could need in a few days for the Auto accept if riot approves it !

**Comment by PhalanXdarKen** [comment]
@Soundofdarkness it will be a pleasure! you just tell me when and how, or if it will be discussed in telegram group etc. I will be there for helping in anything related to translation for my native language Spanish Latam. Soon I will use runebook live on streamings in Twitch so if someone in my audience ask me about, I will tell everyone to give a try to it! so Keep alive that donations integrations / links 😄
